codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -629,15 +634,18 @@ private static ByteBuf encodeProperties(ByteBufAllocator byteBufAllocator,
                         case SERVER_REFERENCE:
                         case REASON_STRING:
                             writeVariableLengthInt(propertiesBuf, property.propertyId);
-                            writeUTF8String(propertiesBuf, ((MqttProperties.StringProperty) property).value);
+                            String value = ((MqttProperties.StringProperty) property).value;
+                            writeUTF8String(propertiesBuf, value, nullableUtf8Bytes(value));
                             break;
                         case USER_PROPERTY:
                             final List<MqttProperties.StringPair> pairs =
                                     ((MqttProperties.UserProperties) property).value;
                             for (MqttProperties.StringPair pair : pairs) {
                                 writeVariableLengthInt(propertiesBuf, property.propertyId);
-                                writeUTF8String(propertiesBuf, pair.key);
-                                writeUTF8String(propertiesBuf, pair.value);
+                                String k = pair.key;
+                                String v = pair.value;
+                                writeUTF8String(propertiesBuf, k, nullableUtf8Bytes(k));
+                                writeUTF8String(propertiesBuf, v, nullableUtf8Bytes(v));
```

Comment:
`maxUtf8Bytes` could be used to "ensure" space for these I think, filling in the length field retroactively

Reply:
Let me rethink this part to handle this one as you suggest (currently  the `writeUTF8String` semantic while using the exact length is a bit different), because it makes sense